actions/service.go

```diff
@@ -74,24 +105,60 @@ func (s *Service) allocateTasks(runID string, actions []*Action) ([]*Task, error
 	return tasks, nil
 }
 
-func (s *Service) runTasks(ctx context.Context, hooks []*Task, event Event, deps Deps) error {
+func (s *Service) runTasks(ctx context.Context, tasks []*Task, event Event, deps Deps) error {
 	var g multierror.Group
-	for _, h := range hooks {
-		hh := h // pinning
+	for _, task := range tasks {
+		task := task // pin
 		g.Go(func() error {
-			return hh.Hook.Run(ctx, event, &HookOutputWriter{
-				RunID:      hh.RunID,
-				ActionName: hh.Action.Name,
-				HookID:     hh.HookID,
+			task.StartTime = time.Now()
+			task.Err = task.Hook.Run(ctx, event, &HookOutputWriter{
+				RunID:      task.RunID,
+				ActionName: task.Action.Name,
+				HookID:     task.HookID,
 				Writer:     deps.Output,
 			})
+			task.EndTime = time.Now()
+			return task.Err
 		})
 	}
 	return g.Wait().ErrorOrNil()
 }
 
-func NewRunID(t time.Time) string {
-	uid := strings.ReplaceAll(uuid.New().String(), "-", "")
-	runID := t.UTC().Format(time.RFC3339) + "_" + uid
-	return runID
+func (s *Service) insertRunInformation(ctx context.Context, event Event, tasks []*Task, endTime time.Time, runErr error) error {
+	_, err := s.DB.Transact(func(tx db.Tx) (interface{}, error) {
+		var err error
+		// insert run information
+		runPassed := runErr == nil
+		_, err = tx.Exec(`INSERT INTO actions_runs(repository_id, run_id, event_type, start_time, end_time, branch_id, source_ref, commit_id, passed)
+			VALUES ($1,$2,$3,$4,$5,$6,$7,'',$8)`,
+			event.RepositoryID, event.RunID.String(), event.EventType, event.EventTime, endTime, event.BranchID, event.SourceRef, runPassed)
+		if err != nil {
+			return nil, fmt.Errorf("insert run information: %w", err)
+		}
+
+		// insert each task information
+		for _, task := range tasks {
+			taskPassed := runErr == nil
```

Comment:
`runErr` is global run error, not the individual task error.

Reply:
good catch, missed this one!